packages/riipen-ui/src/components/MenuList.jsx

```diff
@@ -0,0 +1,166 @@
+    };
+  }
+
+  componentDidMount() {
+    const { autoFocus, children } = this.props;
+    const { activeItemIndex, keyDownHandler } = this.state;
+    window.addEventListener("keydown", keyDownHandler);
+    if (autoFocus && activeItemIndex === -1) {
+      this.moveFocus(children, activeItemIndex, "next");
+    }
+  }
+
+  componentWillUnmount() {
+    const { keyDownHandler } = this.state;
+    window.removeEventListener("keydown", keyDownHandler);
+  }
+
+  getListItems(children, activeItemIndex) {
+    return children.map((child, idx) => {
+      let newProps = {
+        onClick: this.handleClick(child, idx),
+        key: idx,
+        color: this.props.color
+      };
+      if (idx === activeItemIndex) {
+        newProps = Object.assign(newProps, {
+          selected: true
+        });
+      }
+      return React.cloneElement(child, {
+        ...child.props,
+        ...newProps
+      });
+    });
+  }
+
+  handleSelectChange(idx, event) {
+    this.setState({
+      activeItemIndex: idx
+    });
+    const { selectChange } = this.props;
+    if (selectChange) {
+      selectChange(idx, event);
+    }
+  }
+
+  keyDownHandler = event => {
+    const { children } = this.props;
+    const { activeItemIndex } = this.state;
+    const nextEvents = ["ArrowDown", "ArrowRight"];
+    const prevEvents = ["ArrowUp", "ArrowLeft"];
+    const enterEvent = "Enter";
+    const tabEvent = "Tab";
+
+    if (nextEvents.includes(event.key)) {
+      this.moveFocus(children, activeItemIndex, "next", event);
+      event.preventDefault();
+    } else if (prevEvents.includes(event.key)) {
+      this.moveFocus(children, activeItemIndex, "prev", event);
+      event.preventDefault();
+    } else if (event.key === enterEvent) {
+      this.handleSelectChange(this.state.activeItemIndex, event);
+      event.preventDefault();
+    } else if (event.key === tabEvent) {
+      const direction = event.shiftkey ? "prev" : "next";
+      this.moveFocus(children, activeItemIndex, direction, event);
+      event.preventDefault();
+    }
+  };
+
+  moveFocus(listItems, currIndex, direction, event) {
+    let nextIndex = direction === "prev" ? currIndex - 1 : currIndex + 1;
+    if (nextIndex >= listItems.length) {
+      nextIndex = 0;
+    } else if (nextIndex < 0) {
+      nextIndex = listItems.length - 1;
+    }
+
+    const nextItem = listItems[nextIndex];
+    if (!nextItem.props.disabled) {
+      this.handleSelectChange(nextIndex, event);
+      return;
+    }
+    this.moveFocus(listItems, nextIndex, direction);
+    return;
+  }
+
+  handleClick = (child, idx) => {
+    return event => {
+      if (child.props.disabled) return;
+      this.handleSelectChange(idx, event);
+    };
+  };
+
+  render() {
+    const { children, classes } = this.props;
+    const { activeItemIndex } = this.state;
+
+    const listItems = this.getListItems(children, activeItemIndex);
+
+    const className = clsx(classes, "menu-list");
+
+    return (
+      <React.Fragment>
```

Comment:
I don't think you actually need this `<React.Fragment>` here.

Reply:
had styles here at one point but not anymore, good catch